### PR TITLE
Fixed problem with returning inappropriate response statuses for scheduler

### PR DESCRIPTION
**Pull request by ochaparin**

### CODE REVIEW

File: services/self-service/src/main/java/com/epam/dlab/backendapi/dao/ExploratoryDAO.java [hunk 1]
```diff
@@ -129,6 +134,25 @@ public UserInstanceStatus fetchExploratoryStatus(String user, String exploratory
...
 	}
 
+	/**
+	 * Checks status of exploratory. If notebook wasn't found or has one of inappropriate statuses (failed,
+	 * terminated) then throws corresponding exception.
+	 *
+	 * @param user            user name.
+	 * @param exploratoryName the name of exploratory.
+	 */
+	public void checkExploratoryStatusOrElseThrowException(String user, String exploratoryName) {
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/dao/ExploratoryDAO.java]: 1. Checking status may done in mongo query: get user instance only if satisfy some condition (e.g. status != failed | termitated) + TERMINATING. 
2. ChangeMethod return type to user instance.
3. Move user instance state to method parameter end pass it from Service layer.

Alternative solution may be: just use fetchExploratoryStatus and compare it on service layer

File: services/self-service/src/main/java/com/epam/dlab/backendapi/dao/SchedulerJobDAO.java [hunk 2]
```diff
@@ -99,12 +99,9 @@ private Bson statusCondition(UserInstanceStatus desiredStatus) {
...
 	 * @return scheduler job data.
      */
-	public SchedulerJobDTO fetchSingleSchedulerJobByUserAndExploratory(String user, String exploratoryName) {
-		return convertFromDocument((Document) findOne(USER_INSTANCES, and(exploratoryCondition(user, exploratoryName),
-				schedulerNotNullCondition()))
-				.orElseThrow(() -> new DlabException(
-						String.format("Scheduler job for user %s with exploratory instance name %s not found.",
-								user, exploratoryName))).get(SCHEDULER_DATA), SchedulerJobDTO.class);
+	public Optional<SchedulerJobDTO> fetchSingleSchedulerJobByUserAndExploratory(String user, String exploratoryName) {
+		return findOne(USER_INSTANCES, and(exploratoryCondition(user, exploratoryName), schedulerNotNullCondition()))
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/dao/SchedulerJobDAO.java]: Use projection to get only scheduler_data note. map() method in such case will be not needed

File: services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SchedulerJobResource.java [hunk 3]
```diff
@@ -72,19 +71,11 @@ public Response create(@Auth UserInfo userInfo, @PathParam("exploratoryName") St
...
-			return job;
-
-		} catch (Exception t) {
-			log.error("Cannot load scheduler job for user {} and exploratory {} an {}", userInfo.getName(),
-					exploratoryName, t);
-			throw new DlabException("Cannot load scheduler job: " + t.getLocalizedMessage(), t);
-		}
+		SchedulerJobDTO job =
+				schedulerJobService.fetchSchedulerJobForUserAndExploratory(userInfo.getName(), exploratoryName);
+		log.info("Scheduler job data: {}", job);
```
Comment (bhliva) [inline services/self-service/src/main/java/com/epam/dlab/backendapi/resources/SchedulerJobResource.java]: Do we really need it such log message under INFO level?

**Comment by bhliva** [comment]
Looks like controller (SchedulerResource) is not changed but it should

**Comment by ochaparin** [comment]
Accepted

**Pull request merged**